nixos/modules/services/web-apps/ackee.nix

```
@@ -0,0 +1,131 @@
+        <refentrytitle>systemd.exec</refentrytitle><manvolnum>5</manvolnum>
+        </citerefentry>.
+
+        This file should be owned by root and not be readable by anyone but root.
+        The file complements the <option>settings</option> option and is supposed to contain
+        confidential settings like <literal>ACKEE_PASSWORD</literal>.
+
+        <programlisting>
+          # content of the environment file
+          ACKEE_PASSWORD=verysecretpassword
+        </programlisting>
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    # Warn on outdated mongodb
+    warnings = mkIf
+      (cfg.settings.ACKEE_MONGODB == "mongodb://localhost:27017/ackee"
+        && config.services.mongodb.enable == true
+        && versionOlder config.services.mongodb.package.version "4.0.6") [
+      "Ackee needs at least mongodb 4.0.6"
+    ];
+
+    services.mongodb = mkIf (cfg.settings.ACKEE_MONGODB == defaultSettings.ACKEE_MONGODB) {
+      enable = mkDefault true;
+      package = mkDefault pkgs.mongodb-4_2;
+    };
+
+    # Default config
+    services.ackee.settings = mapAttrs (_: mkDefault) defaultSettings;
+
+    systemd.services.ackee = {
+      description = "Ackee Service";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "networking.target" "mongodb.service" ];
+      environment = cfg.settings;
+
+      serviceConfig = {
+        WorkingDirectory = "%S/ackee";
+        StateDirectory = "ackee";
+        StateDirectoryMode = "0700";
+        ExecStart = "${pkgs.ackee}/bin/ackee";
+        EnvironmentFile = cfg.environmentFile;
+
+        AmbientCapabilities = "CAP_NET_BIND_SERVICE";
+        BindReadOnlyPaths = [
+          "/nix/store"
+          "-/etc/resolv.conf"
+          "-/etc/nsswitch.conf"
+          "-/etc/hosts"
+          "-/etc/localtime"
+        ];
+        CapabilityBoundingSet = "CAP_NET_BIND_SERVICE";
+        # ProtectClock= adds DeviceAllow=char-rtc r
+        DeviceAllow = "";
+        DynamicUser = true;
+        LockPersonality = true;
+        # node.js needs this
+        #MemoryDenyWriteExecute = true;
+        NoNewPrivileges = true;
+        PrivateDevices = true;
+        PrivateMounts = true;
+        # Port needs to be exposed to the host network
+        #PrivateNetwork = true;
+        PrivateTmp = true;
+        PrivateUsers = true;
+        ProcSubset = "pid";
+        ProtectClock = true;
+        ProtectHome = true;
+        ProtectHostname = true;
+        ProtectSystem = "strict";
+        ProtectControlGroups = true;
+        ProtectKernelLogs = true;
+        ProtectKernelModules = true;
+        ProtectKernelTunables = true;
+        ProtectProc = "invisible";
+        RestrictAddressFamilies = [ "AF_INET" "AF_INET6" ];
+        RestrictNamespaces = true;
+        RestrictRealtime = true;
+        RestrictSUIDSGID = true;
+        SystemCallArchitectures = "native";
+        SystemCallFilter = [ "@system-service" "~@privileged @resources @setuid @keyring" ];
```

I think *either*:

```suggestion
        SystemCallFilter = [ "@system-service" "~@privileged" "@resources" "@setuid" "@keyring" ];
```

or make it one single space-separated string. Don't mix them :P

---

There is a `~` in front of the second list, so all system calls in `@system-service` are allowed, except those in the second list :)